### ISSUE TITLE
Better Web Mode format detection

### ISSUE THREAD

**Issue by ferrum-salvator** [lassik/emacs-format-all-the-code#14]
Web Mode is currently always interpreted as XML, if I read the source correctly, which is very wrong.  I propose that with multi-language major modes minor modes be determined by regex. If used as a fallback as well, this will help with lesser-used major modes, as well.

**Comment by lassik** [comment]
I think we should just hard-code the mode lists. It takes only a minute to add a mode to the list - just commit the change and push to GitHub. MELPA picks it up in a few hours. Always better to avoid regexp hacks (and heuristics in general) where possible.

I tried formatting a HTML file containing a `<br>` tag (no closing tag - which is valid HTML but invalid XML) in `web-mode` and `tidy` seems to handle it fine as HTML. What part of the code do you think would cause `web-mode` to be treated as XML? The [`:modes` listing](https://github.com/lassik/emacs-format-all-the-code/blob/master/format-all.el#L405) is just a set of modes to format using HTML Tidy. Despite its name, HTML Tidy handles both XML and HTML (up-to-date versions also handle HTML5).

**Comment by lassik** [comment]
Actually, the code had the opposite bug :D Tidy would *never* be invoked in XML mode, even when the `major-mode` is `xml-mode` or `nxml-mode`. I didn't realize it needs the `-xml` flag to be given manually. Fixed.

**Comment by lassik** [comment]
Apparently Tidy auto-detects that your file is XHTML (as opposed to HTML) when you have the right [doctype declaration](https://www.w3.org/QA/2002/04/valid-dtd-list.html) in your file.

**Comment by ferrum-salvator** [comment]
Yes, but I was use web-mode to edit JS and JSX files - I prefer their comment implementation, for one. That's what I meant - some modes are ambiguous, and might need different formatters.

Anton Shiryaev
Space Monitoring Data Center SINP MSU
anton.o.shiryaev@gmail.com

**Comment by lassik** [comment]
I see. I'll have to write some code to look at `web-mode-content-type` and whatever variable `web-mode` uses to indicate the template engine. `prettier` would be formatter of choice for JSX, as it is now when using `jsx-mode`.
